molecule.go

```diff
@@ -242,18 +242,20 @@ func (m Molecule) Mount(dest string) error {
 			return err
 		}
 
-		workdir := filepath.Join(metadir, "work")
-		if err := EnsureDir(workdir); err != nil {
-			return errors.Wrapf(err, "failed to ensure workdir")
+		persistMetaPath := m.config.WriteableOverlayPath
+		if persistMetaPath == "" {
+			// no configured path, use metadir
+			persistMetaPath = metadir
 		}
 
-		upperdir := m.config.WriteableOverlayPath
-		if upperdir == "" {
-			upperdir = filepath.Join(metadir, "persist")
+		workdir := filepath.Join(persistMetaPath, "work")
+		if err := EnsureDir(workdir); err != nil {
+			return errors.Wrapf(err, "failed to ensure workdir %q", workdir)
 		}
 
+		upperdir := filepath.Join(persistMetaPath, "persist")
```

Comment:
Hm, this kinda sucks, but this is not quite in keeping with the intent of --persist as used by lxc.  Now it probably doesn't matter - users will know where to find $path/persist, and lxc will always just pass the same dirname to --persist.

So what you have here is probably the best we can do.